### PR TITLE
Cell void fraction test

### PR DESCRIPTION
**Pull request by voferreira**
# Description of the problem

- Void fraction used to calculate the void fraction applied in drag models was always the interpolated one.
- We would like to test the cell void fraction instead

# Description of the solution

- Created a variable in the scratch data to store the void fraction of each cell
- Changed the name of the interpolated void fraction to 'interpolated_void_fraction'
- Created a parameter to control whether the void fraction used to calculate drag is the cell one or the interpolated at the particle's position

# How Has This Been Tested?

- Sedimentation of a single particle.

# Documentation

- [ ] Unresolved CFD-DEM parameters guide

### CODE REVIEW

**Review by blaisb — CHANGES_REQUESTED** [review]
This is a good implementation, but i think there is an easier solution. The issue I have with this current implementation is that every new drag model will require to take this if condition into account. I think it might be better to store the void_fraction for each particle in the reinit differently depending on this switch. If this switch is true, then you store the interpolated void fraction. if it's false, then you store the constant void fraction (but still in a vector). Thus, every force model will see the new void fraction model no matter what is the implementation. This will be easier to maintain in the long term. We can talk about it tomorrow if you want :)!

**Review by shahabgol — APPROVED** [review]

File: source/fem-dem/parameters_cfd_dem.cc [hunk 1]
```diff
@@ -80,6 +80,11 @@ namespace Parameters
                       "true",
                       Patterns::Bool(),
                       "Choose whether or not to implement d(epsilon)/dt ");
+    prm.declare_entry(
+      "interpolated void fraction",
+      "true",
```
[the remark this answers is not in the document]
Comment (voferreira) [reply source/fem-dem/parameters_cfd_dem.cc:85]: The non interpolated one is a test. The original implementation uses the interpolated one.

File: source/fem-dem/parameters_cfd_dem.cc [hunk 2]
```diff
@@ -80,6 +80,11 @@ namespace Parameters
...
+      Patterns::Bool(),
+      "Choose whether the void fraction is the one of the cell or the interpolated at the particle position.");
```
Comment (tonielgeitani) [inline source/fem-dem/parameters_cfd_dem.cc:87]:
```suggestion
      "Choose whether the void fraction is the one of the cell or the one interpolated at the particle position.");
```

**Review by tonielgeitani — APPROVED** [review]

File: source/solvers/navier_stokes_scratch_data.cc [hunk 3]
```diff
@@ -179,7 +179,9 @@ NavierStokesScratchData<dim>::enable_particle_fluid_interactions(
     std::vector<Tensor<1, dim>>(n_global_max_particles_per_cell);
   fluid_velocity_at_particle_location =
     std::vector<Tensor<1, dim>>(n_global_max_particles_per_cell);
-  cell_void_fraction = std::vector<double>(n_global_max_particles_per_cell);
+  cell_void_fraction = double(n_global_max_particles_per_cell);
```
[the remark this answers is not in the document]
Comment (tonielgeitani) [reply source/solvers/navier_stokes_scratch_data.cc]: This line should be defined the same way you define the interpolated_void_fraction as:
cell_void_fraction =
     std::vector<double>(n_global_max_particles_per_cell);

**Review by blaisb — CHANGES_REQUESTED** [review]
One last tiny thing to change. Good work.

**Review by blaisb — APPROVED** [review]

**Pull request merged**